File: app/src/test/java/com/example/carlauncher/data/DataStatusTest.java
package com.example.carlauncher.data;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * DataStatus 枚举与协议线值的映射契约。
 */
public class DataStatusTest {

    @Test
    public void getValue_matchesWireContract() {
        assertEquals(0, DataStatus.NORMAL.getValue());
        assertEquals(1, DataStatus.INVALID.getValue());
        assertEquals(2, DataStatus.NO_DATA.getValue());
        assertEquals(3, DataStatus.SOURCE_DISCONNECTED.getValue());
        assertEquals(4, DataStatus.TRANSPORT_DISCONNECTED.getValue());
    }
}

File: app/src/test/java/com/example/carlauncher/data/DataStatusTest.kt
package com.example.carlauncher.data

import org.junit.Assert.assertEquals
import org.junit.Test

/** DataStatus 枚举与协议线值的映射契约。 */
open class DataStatusTest {

    @Test
    open fun getValue_matchesWireContract() {
        assertEquals(0, DataStatus.NORMAL.value.toLong())
        assertEquals(1, DataStatus.INVALID.value.toLong())
        assertEquals(2, DataStatus.NO_DATA.value.toLong())
        assertEquals(3, DataStatus.SOURCE_DISCONNECTED.value.toLong())
        assertEquals(4, DataStatus.TRANSPORT_DISCONNECTED.value.toLong())
    }
}
